metals/src/main/scala/scala/meta/internal/metals/debug/DebugProvider.scala

```diff
@@ -97,13 +106,16 @@ class DebugProvider(
   def resolveMainClassParams(
       params: DebugUnresolvedMainClassParams
   )(implicit ec: ExecutionContext): Future[b.DebugSessionParams] = {
-    withRebuildRetry(() =>
+    val result = withRebuildRetry(() =>
       buildTargetClassesFinder
         .findMainClassAndItsBuildTarget(
           params.mainClass,
           Option(params.buildTarget)
         )
     ).map {
+      case (clazz, target) :: others
+          if buildClient.buildHasErrors(target.getId()) =>
+        throw WorkspaceErrorsException
```

Comment:
why not return Future.failed(WorkspaceErrorsException) and flatMap before ?

Reply:
I would need to add Future() to the cases before and this does basically the same thing.

Reply:
you can add Future.successful for the others. 
Isn't throwing and catching exceptions costly?

Reply:
I don't think there is much difference. But I guess using Future.failed is more idiomatic. I will change.